src/github.com/matrix-org/dendron/dendron/main.go

```diff
@@ -148,49 +125,62 @@ func main() {
 		}
 	}
 
-	var synapse *exec.Cmd
-	var pusher *exec.Cmd
-	var synchrotron *exec.Cmd
-
 	var synapseLog = log.WithField("synapse", synapseURL.String())
 
+	// Used to terminate dendron.
+	terminate := make(chan string, 1)
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		s := <-signals
+		terminate <- fmt.Sprint("Got signal", s)
+	}()
+
 	if *startSynapse {
-		synapse = exec.Command(*synapsePython, "-m", "synapse.app.homeserver", "-c", *synapseConfig)
+		synapse := exec.Command(*synapsePython, "-m", "synapse.app.homeserver", "-c", *synapseConfig)
 		synapse.Stderr = os.Stderr
 		synapseLog.Print("Starting synapse")
 
 		synapse.Start()
-
-		sh := signalHander{
-			synapseLog: synapseLog,
-			synapse:    synapse.Process,
-		}
-		channel := make(chan os.Signal, 1)
-		signal.Notify(channel, os.Interrupt)
-		go sh.handleSignal(channel)
+		defer synapse.Process.Signal(os.Interrupt)
 
 		if err := waitForSynapse(synapseURL, synapseLog); err != nil {
 			synapseLog.Panic(err)
 		}
 
+		go func() {
+			synapse.Process.Wait()
+			terminate <- "Synapse Stopped"
+		}()
+
 		if *pusherConfig != "" {
-			pusher = exec.Command(*synapsePython, "-m", "synapse.app.pusher", "-c", *pusherConfig)
+			pusher := exec.Command(*synapsePython, "-m", "synapse.app.pusher", "-c", *pusherConfig)
 			pusher.Stderr = os.Stderr
 			synapseLog.Print("Starting pusher")
 			pusher.Start()
-			sh.pusher = pusher.Process
+			defer pusher.Process.Signal(os.Interrupt)
```

Comment:
You're dropping any `error` return value to the floor: https://golang.org/pkg/os/#Process.Signal
You may need to make an anonymous function to handle this, e.g:
```go
defer func() {
  err := pusher.Process.Signal(os.Interrupt)
  if err != nil {
    log.WithError(err).Print("Failed to signal pusher")
  }
}()
```